### ISSUE TITLE
Feat(frontend): signup flow

### ISSUE THREAD

**Issue by kittybest** [privacy-scaling-explorations/maci-rpgf#4]
As a user, I want to be able to sign up w/ MACI so that I can vote

## Requirements
- [x] add ability to connect wallet (this is already provided by EasyRetroPGF)
- [x] generate MACI key for user
- [x] execute MACI `signUp` function (this should pass relevant data to EAS gatekeeper)

### EAS attestation

Currently the app relies on the `approval` schema to validate if somone can vote. We should update this to use the `badgeholder` schema

When a user connects their wallet, we should query EAS to confirm if the address is an OP badgeholder or not
- If yes, we should allow them to sign up 
- If not, we should disable the ability to sign up or vote (and display that in the UI)
   - For this, maybe it's as simple as disabling the "sign up" button & saying "You must be a badgeholder to sign up"

**Note: EasyRetroPGF may already do this check (after they use SIWE to connect?), so let's check out the code & confirm before implementing ourselves**

### MACI key generation

Once a user connects their wallet & we've confirmed they're an OP badgeholder (by querying the EAS attestation), we should generate a MACI keypair for them so they can sign up

Let's generate the MACI key deterministically, so it's attached to their ETH address (so same implementation as clr.fund). This way, we can completely abstract the MACI keypair away from the user. They don't even need to know it exists. As long as they can connect their wallet & prove ownership of the ETH address, they should be able to generate their MACI private key to sign any messages.

### Signup flow

This is a new flow to the EasyRetroPGF app... so I'm thinking this simplest/easiest implementation is this...

Once wallet is connected, allow users to add projects to their ballot (so same UX as current app)

In their ballot view, instead of allowing users to submit ballot, first prompt them to sign up:

![Image 2024-01-29 at 13 13 44](https://github.com/privacy-scaling-explorations/maci-rpgf/assets/8097623/ccbfa4f5-769e-4adf-a5a6-375556ea2dff)

So basically: change the text of the button (to "Sign up") & change the tx this triggers - this should call the `signup` function from the MACI contract. Once this tx is submitted & succeeds, then we should change the text back to "Submit ballot".

**Comment by kittybest** [comment]
> ![Image 2024-01-29 at 13 13 44](https://private-user-images.githubusercontent.com/8097623/300577932-ccbfa4f5-769e-4adf-a5a6-375556ea2dff.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY1ODYyMjQsIm5iZiI6MTcwNjU4NTkyNCwicGF0aCI6Ii84MDk3NjIzLzMwMDU3NzkzMi1jY2JmYTRmNS03NjllLTRhZGYtYTVhNi0zNzU1NTZlYTJkZmYuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMzBUMDMzODQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTE1YzU1NmZjYWNjZTVlYzczYmYwZGQwYzdmMzZmMWJjZjE5Y2Y0Zjk4NGE4YjU3MTlhMzcxNmZiYzM4OWI1MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PTY7yfR1AiCfjgV5VC5C6tRsNRWS4y4XJyBO-kr1yYU)
> 
> So basically: change the text of the button (to "Sign up") & change the tx this triggers - this should call the `signup` function from the MACI contract. Once this tx is submitted & succeeds, then we should change the text back to "Submit ballot".

Question: so we still allow them to add projects to ballot even if they haven't signed up? I suppose they won't see that page until they already signed up.

**Comment by samajammin** [comment]
Solved by #20 #23 #30 https://github.com/privacy-scaling-explorations/maci/pull/1212
